aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java

```diff
@@ -567,15 +567,19 @@ private static void compileCallIndirect(MethodVisitor asm, int typeId, FunctionT
         int slot = type.params().stream().mapToInt(AotUtil::slotCount).sum();
 
         // parameters: arguments, funcTableIdx, tableIdx, instance
-        emitBoxArguments(asm, type.params());
+        // emitBoxArguments(asm, type.params());
+        // asm.visitLdcInsn(LONG);
+        for (int i = 0; i < type.params().size(); i++) {
+            asm.visitVarInsn(Opcodes.LLOAD, i);
+        }
         asm.visitLdcInsn(typeId);
         asm.visitVarInsn(Opcodes.ILOAD, slot); // funcTableIdx
         asm.visitVarInsn(Opcodes.ILOAD, slot + 1); // tableIdx
         asm.visitVarInsn(Opcodes.ALOAD, slot + 2); // instance
 
         emitInvokeStatic(asm, AotMethods.CALL_INDIRECT);
 
-        emitUnboxResult(type, asm);
```

Comment:
I'm a bit tired now, but I leave this comment anyhow to see the reaction :-)

What if we change all the signatures of the methods `@UsedByGeneratedCode` and the intrinsics to deal with solely `long` types as they are returned from the stack?

Pros:
- having to deal only with the arity we can simplify the code of the aot
- we keep the conversions in plain Java code
- it's easier to reason about
Cons:
- performance? I have no idea of the impact, it would be great to reason on numbers but I would listen to gut feelings

Reply:
I don’t think that’s feasible. You would need to convert to and from native types for every operation. This would likely be slower and would bloat the byte code. It’s actually more complicated in many ways than using native types.

Reply:
thanks for the feedback!